mne/channels/channels.py

```diff
@@ -575,10 +575,26 @@ def drop_channels(self, ch_names):
         -----
         .. versionadded:: 0.9.0
         """
-        bad_idx = [self.ch_names.index(c) for c in ch_names
-                   if c in self.ch_names]
+        msg = ("'ch_names' should be a list of strings (the name[s] of the "
+               "channel to be dropped), not a {0}.")
+        if isinstance(ch_names, string_types):
+            raise TypeError(msg.format("string"))
+        else:
+            if not all([isinstance(ch_name, string_types)
+                        for ch_name in ch_names]):
+                raise TypeError(msg.format(type(ch_names[0])))
+
+        missing = [ch_name for ch_name in ch_names
+                   if ch_name not in self.ch_names]
+        if len(missing) > 0:
+            msg = "Channel(s) {0} not found, cannot drop."
```

Comment:
I would say ``"Channel(s) {0} not found, nothing dropped."`` as even the channels that were found will not be dropped.